components/examples/await/src/main.ts

```diff
@@ -0,0 +1,54 @@
+import van from 'vanjs-core';
+import { Await } from 'vanjs-ui';
+
+const { div, h2, span, button } = van.tags;
+
+async function fetchWithDelay<Result>(
+  url: string,
+  wait: number
+): Promise<Result> {
+  const delay = (ms: number) =>
+    new Promise<void>((resolve) => {
+      setTimeout(() => resolve(), ms);
+    });
+  await delay(wait);
+  const response = await fetch(url);
+  return response.json();
+}
+
+const fetchStar = () =>
+  fetchWithDelay<{ stargazers_count: number }>(
+    'https://api.github.com/repos/vanjs-org/van',
+    1000
+  ).then((data) => data.stargazers_count);
+
+function App() {
+  const data = van.state(fetchStar());
+
+  return [
+    () =>
+      h2(
+        'Github Star: ',
+        Await(
+          {
+            value: data.val,
+            Loading: () => `🌀 Loading...`,
+            Error: () => `🙀 Request failed.`,
+          },
+          (starNumber) => span(`🎉 ${starNumber}!`)
+        )
+      ),
+    () =>
+      div(
+        Await(
+          {
+            value: data.val,
+            Loading: () => '',
```

Comment:
Must return an non-null value, otherwise the children will not be rendered even the status has changed to  fulfilled. Can anyone help me with this?

Reply:
Yes. This is working as intended. In a `State`-derived DOM node, returning `null` or `undefined` in the binding function means you want to delete the node. See https://vanjs.org/tutorial#removing-a-dom-node.